text/src/autogluon/text/automm/data/collator.py

```diff
@@ -0,0 +1,324 @@
+import math
+import warnings
+import numpy as np
+import torch
+
+__all__ = ["Stack", "Pad", "Tuple", "List", "Dict", "PadImage", "Zip"]
+
+
+def _pad_arrs_to_max_length(arrs, pad_axis, pad_val, round_to=None, max_length=None):
+    """Inner Implementation of the Pad collate
+
+    Arguments:
+        arrs (list)
+        pad_axis (int)
+        pad_val (number)
+        round_to (int, optional). (default: ``None``)
+        max_length (int, optional). (default: ``None``)
+
+    Returns:
+        ret : torch.Tensor
+        original_length : torch.Tensor
+    """
```

Comment:
Transformers don't where the padding is as long as we give it the correct mask. Here we pad the right side since our masks are computed based on the valid length.